lib/jobs/pdu-node-relations.js

```diff
@@ -0,0 +1,142 @@
+// Copyright 2016, EMC, Inc.
+
+'use strict';
+
+var di = require('di');
+
+module.exports = pduRelationsJobFactory;
+di.annotate(pduRelationsJobFactory, new di.Provide('Job.Catalog.PduRelations'));
+di.annotate(pduRelationsJobFactory, new di.Inject(
+    'Job.Base',
+    'Services.Waterline',
+    'Logger',
+    'Util',
+    'Promise',
+    'Assert'
+));
+
+function pduRelationsJobFactory(
+    BaseJob,
+    waterline,
+    Logger,
+    util,
+    Promise,
+    assert
+) {
+
+    var logger = Logger.initialize(pduRelationsJobFactory);
+
+    /**
+     *
+     * @param {Object} [options]
+     * @constructor
+     */
+    function PduRelationsJob(options, context, taskId) {
+        PduRelationsJob.super_.call(this, logger, options, context, taskId);
+
+        this.nodeId = context.target || options.nodeId;
+        assert.isMongoId(this.nodeId);
+    }
+
+    util.inherits(PduRelationsJob, BaseJob);
+
+    /**
+     * Globals
+     */
+
+    var RELATION_TYPE = 'poweredBy';
+
+    /**
+     * @memberOf PduRelationsJob
+     */
+    PduRelationsJob.prototype._run = function run() {
+        var self = this;
+        return waterline.nodes.findByIdentifier(self.nodeId)
+        .then(function (node) {
+            if (node.type !== 'pdu') {
+                throw new Error('Task runs only on PDU nodes');
+            }
+            var pduHost = _adjustIpStrFormat(node.snmpSettings.host);
+            return self._searchObmServices(pduHost);
+        })
+        .then(function () {
+            self._done();
+        })
+        .catch(function (err) {
+            self._done(err);
+        });
+
+    };
+
+    PduRelationsJob.prototype._searchObmServices =
+        function _searchObmServices(pduHost) {
+            var self = this;
+            return Promise.try(function () {
+                return waterline.obms.find();
+            })
+            .filter(function (obm) {
+                return (_validateIpAddressFormat(obm.config.host));
+            })
+            .map(function (result) {
+                //compare hosts
+                var obmServiceHost = _adjustIpStrFormat(result.config.host);
+                if (pduHost === obmServiceHost) {
+                    //find the node
+                    return waterline.nodes.findByIdentifier(result.node)
+                   .then(function (node) {
+                       if (!node) {
+                           throw new Error('Could not find node with identifier ' + result.node);
+                       }
+                       return node.relations;
+                   })
+                   .filter(function (entry) {
+                       return (entry.targets.indexOf(self.nodeId) < 0);
+                   })
+                   .then(function (relations) {
+                       relations.push({
+                           relationType: RELATION_TYPE,
```

Comment:
I think you are copying the design of switch-node-relation job, but it still doesn't resolve the bi-direction relations, so it still need some fix.

The relation is always bi-direction, if you relates A with B by "poweredBy" and update the A's document, then you should still relates B with A by "powers" and update B's document. 

I admit this is a little complicated, but this is exactly the design with enclosure relation. Either build or remove relation is not easy.

I just come up an idea, copy the current `obm` design, detach the relations to a new kind of resource, this will simplify a lot code logic to either build or remove relations.